include/zephyr/bluetooth/services/ots.h

```diff
@@ -725,6 +726,24 @@ struct bt_ots_cb {
 	 */
 	void (*obj_name_written)(struct bt_ots *ots, struct bt_conn *conn,
 				 uint64_t id, const char *cur_name, const char *new_name);
+
+	/** @brief Object Calculate checksum callback
+	 *
+	 *  This callback is called when the OACP Calculate Checksum procedure is performed.
+	 *  Because object data is opaque to OTS, the application is the only one who
+	 *  knows where data is and should return pointer of actual object data.
+	 *
+	 *  @param[in]	ots      OTS instance.
+	 *  @param[in]	conn     The connection that wrote object.
+	 *  @param[in]	id       Object ID.
+	 *  @param[in]	offset   The first octet of the object contents need to be calculated.
+	 *  @param[in]	len      The length number of octets object name.
+	 *  @param[out] data     Pointer of actual object data.
+	 *
+	 *  @return 0 to accept, or any negative value to reject.
+	 */
+	int (*obj_cal_checksum)(struct bt_ots *ots, struct bt_conn *conn, uint64_t id,
```

Comment:
Can't we just use the `obj_read` method?

Reply:
Considering obj_read might fragment data. like what we do in [sample ](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/peripheral_ots/src/main.c#L152).
I think it's better to have a dedicate callback for checksum,